### ISSUE TITLE
In windows the pause/stop fit option is greyed out once fit is started

### ISSUE THREAD

**Issue by acaruana2009** [easyScience/easyReflectometryApp#35]
Once a fit is started it cannot be actively stopped. For some reason in windows, the pause fit option is greyed out. I have yet to be able to try this on linux to check if this is the case here as well.

**Comment by arm61** [comment]
Yeah, this is an artifact of the current threading methodology (which is under investigation to enable this in future). As a short term fix, do you think it would be a better UI if instead of the pause logo and it reading "fitting in progress", there should be a spinner (like this https://fontawesome.com/v5.15/icons/spinner?style=solid) to indicate that it is in progress?

**Comment by AndrewSazonov** [comment]
In easyDiffraction, an animated message "Fitting in progress..." is displayed in the status bar (bottom right corner) to indicate that fitting is in progress.

**Comment by arm61** [comment]
I believe that is the case in easyReflectometry also (https://github.com/easyScience/easyReflectometryApp/blob/48f18dd63f6f3bbc3ad87d5460925a3b6770ae4a/easyReflectometryApp/Gui/Pages/Analysis/SideBarBasic.qml#L195). However, I think the `pause-circle` glyph is potentially confusing.

**Comment by AndrewSazonov** [comment]
No, this animated message "Fitting in progress..." is implemented based on the RunningLabel.qml element from `easyApp`, which was not available in `easyAppGui`. So, this is yet another reason to switch from the old `easyAppGui` and `easyAppLogic` to the new `easyApp` as early as possible (Issue #19). 

Meanwhile, if the `pause-circle` glyph is confusing, I'd suggest simply disabling the `Start fitting` button on click, without changing its text and icon, until the pause/stop function is implemented (See PR #36).

**Comment by arm61** [comment]
Not quite resolved with #36, however a short term solution has been implemented.

**Comment by arm61** [comment]
@AndrewSazonov or @rozyczko is it now possible in easyDiffraction to pause/stop the fitting process?

**Comment by rozyczko** [comment]
fit thread cancellation was implemented and worked fine on windows but not on macos. There's a bug in macos version of Qt which makes thread stopping affect the main thread and make it unstable.
I think we decided to wait for qt to fix their issue.
Maybe we can try it with Qt6 at some point?
